tests/by-util/test_expr.rs

```diff
@@ -116,6 +131,94 @@ fn test_parenthesis() {
         .stderr_only("expr: syntax error: unexpected argument '('\n");
 }
 
+#[test]
+fn test_missing_parenthesis() {
+    new_ucmd!()
+        .args(&["(", "2"])
+        .fails()
+        .code_is(2)
+        .stderr_only("expr: syntax error: expecting ')' after '2'\n");
+
+    new_ucmd!()
+        .args(&["(", "2", "a"])
+        .fails()
+        .code_is(2)
+        .stderr_only("expr: syntax error: expecting ')' instead of 'a'\n");
+}
+
+#[test]
+fn test_regexp_unmatched() {
+    new_ucmd!()
+        .args(&["_", ":", "a\\("])
+        .fails()
+        .code_is(2)
+        .stderr_only("expr: Unmatched ( or \\(\n");
+
+    new_ucmd!()
+        .args(&["_", ":", "a\\)"])
+        .fails()
+        .code_is(2)
+        .stderr_only("expr: Unmatched ) or \\)\n");
+
+    new_ucmd!()
+        .args(&["_", ":", "a\\{1"])
+        .fails()
+        .code_is(2)
+        .stderr_only("expr: Unmatched \\{\n");
+
+    new_ucmd!()
+        .args(&["_", ":", "a\\}1"])
+        .fails()
+        .code_is(2)
+        .stderr_only("expr: Unmatched \\}\n");
+
+    new_ucmd!()
+        .args(&["_", ":", "a\\{1a\\}"])
+        .fails()
+        .code_is(2)
+        .stderr_only("expr: Invalid content of \\{\\}\n");
+
+    new_ucmd!().args(&["a", ":", "\\(b\\)*"]).fails().code_is(1);
+    // TODO bring back
+    // .no_output();
+}
+
+#[test]
+fn test_checks() {
+    new_ucmd!()
+        .args(&["a\nb", ":", "a\\$"])
+        .fails()
+        .code_is(1)
+        .stdout_only("0\n");
+
+    new_ucmd!()
+        .args(&["a(", ":", "a("])
+        .succeeds()
+        .stdout_only("2\n");
+
+    new_ucmd!()
+        .args(&["a(", ":", "a("])
+        .succeeds()
+        .stdout_only("2\n");
+
+    new_ucmd!()
+        .args(&["_", ":", "a\\{1,0\\}"])
+        .fails()
+        .code_is(2)
+        .stderr_only("expr: Invalid content of \\{\\}\n");
+
+    new_ucmd!()
+        .args(&["_", ":", "a\\{32768\\}"])
+        .fails()
+        .code_is(2)
+        .stderr_only("expr: Invalid content of \\{\\}\n");
```

Comment:
Is the error message deliberately different from GNU `expr` ("expr: Regular expression too big")?